cpp/betweenness_centrality_module/algorithm_online/betweenness_centrality_online.cpp

```diff
@@ -0,0 +1,316 @@
+      this->node_bc_scores[w_id] += dependency_s_on.at(w_id) / quotient;
+    }
+
+    if (affected_bcc_articulation_points.count(s_id)) {
+#pragma omp atomic update
+      this->node_bc_scores[w_id] += dependency_s_on.at(w_id) * peripheral_subgraphs_order.at(s_id);
+#pragma omp atomic update
+      this->node_bc_scores[w_id] += ext_dependency_s_on.at(w_id) / quotient;
+    }
+  }
+}
+
+void OnlineBC::iCentral(const mg_graph::GraphView<> &prior_graph, const mg_graph::GraphView<> &current_graph,
+                        const Operation operation, const std::pair<std::uint64_t, std::uint64_t> updated_edge,
+                        const std::uint64_t threads) {
+  const mg_graph::GraphView<> &graph_with_updated_edge =
+      (operation == Operation::INSERT_EDGE) ? current_graph : prior_graph;
+
+  std::unordered_set<std::uint64_t> articulation_points_by_bcc;
+  std::vector<std::unordered_set<std::uint64_t>> nodes_by_bcc;
+  std::vector<std::vector<mg_graph::Edge<>>> edges_by_bcc;
+  edges_by_bcc =
+      bcc_algorithm::GetBiconnectedComponents(graph_with_updated_edge, articulation_points_by_bcc, nodes_by_bcc);
+
+  const auto [affected_bcc_nodes, affected_bcc_articulation_points] =
+      IsolateAffectedBCC(graph_with_updated_edge, updated_edge);
+
+  const auto distances_first = SSSPLengths(graph_with_updated_edge, updated_edge.first, affected_bcc_nodes);
+  const auto distances_second = SSSPLengths(graph_with_updated_edge, updated_edge.second, affected_bcc_nodes);
+
+  const auto peripheral_subgraphs_order =
+      PeripheralSubgraphsOrder(prior_graph, affected_bcc_articulation_points, affected_bcc_nodes);
+
+  // OpenMP might throw errors when iterating over STL containers
+  auto array_size = affected_bcc_nodes.size();
+  std::uint64_t affected_bcc_nodes_array[array_size];
+  std::uint64_t i = 0;
+  for (const auto node_id : affected_bcc_nodes) {
+    affected_bcc_nodes_array[i] = node_id;
+    i++;
+  }
+
+  omp_set_dynamic(0);
+  omp_set_num_threads(threads);
+#pragma omp parallel for
+  for (std::uint64_t i = 0; i < array_size; i++) {
+    auto node_id = affected_bcc_nodes_array[i];
+    if (distances_first.at(node_id) != distances_second.at(node_id))
+      Iteration(prior_graph, current_graph, node_id, affected_bcc_nodes, affected_bcc_articulation_points,
+                peripheral_subgraphs_order);
+  }
+}
+
+std::unordered_map<std::uint64_t, double> OnlineBC::Set(const mg_graph::GraphView<> &graph, const bool directed,
+                                                        const bool normalize, const std::uint64_t threads) {
+  this->directed = directed;
+
+  BrandesWrapper(graph, threads);
+  this->computed = true;
+
+  if (normalize) return NormalizeBC(this->node_bc_scores, graph.Nodes().size());
+
+  return this->node_bc_scores;
+}
+
+std::unordered_map<std::uint64_t, double> OnlineBC::Get(const mg_graph::GraphView<> &graph, const bool normalize) {
+  if (!this->computed) BrandesWrapper(graph, std::thread::hardware_concurrency());
```

Comment:
Can you repeat, what are the steps to reproduce this?

Reply:
`Get` is never called without initialization by the query module (it does the checking), and I'll thus remove this check and the relevant unit test (`TestGetUninitialized`) as unnecessary.